### ISSUE TITLE
Diagnostic message for creating a strand with a `void` function is ambiguous

### ISSUE THREAD

**Issue by IMS94** [ballerina-platform/ballerina-lang#30325]
**Description:**
Consider the following code snippet:
```
future f1 = start testAsync();
```
where the function is:
```
function testAsync() {
    
}
```
Here we get an error and the diagnostic message shown is `incompatible types: expected 'future', found 'future'(BCE2066)`. This error message is not informative enough.

**Steps to reproduce:**
See description

**Affected Versions:**
2.0.0-alpha9-SNAPSHOT

**Suggested Labels (optional):**
Compiler Frontend, Diagnostics

**Comment by pubudu91** [comment]
This looks like a bug in the parser. According to the spec, `future` is a valid type descriptor. But the parser gives syntax errors when trying to use `future` without the type parameter. I guess the unclear error message is due to the structure of the recovered tree.
https://ballerina.io/spec/lang/master/#future-type-descriptor

**Comment by lochana-chathura** [comment]
Related to #28758

**Comment by lochana-chathura** [comment]
With parsing issue being fixed in #30678, following compiles with no errors. Hence closing this issue.
```bal
public function main() {
    future f1 = start testAsync();
}


function testAsync() {   
}
```
